core/util/src/mill/util/Jvm.scala

```diff
@@ -620,12 +620,23 @@ object Jvm {
       id: String,
       ctx: Option[mill.define.TaskCtx] = None,
       coursierCacheCustomizer: Option[FileCache[Task] => FileCache[Task]] = None,
-      jvmIndexVersion: String = mill.api.BuildInfo.coursierJvmIndexVersion
+      jvmIndexVersion: String = mill.api.BuildInfo.coursierJvmIndexVersion,
+      useShortPaths: Boolean = false
   ): Result[os.Path] = {
     val coursierCache0 = coursierCache(ctx, coursierCacheCustomizer)
+    val shortPathDirOpt = Option.when(useShortPaths) {
+      if (isWin)
+        os.Path(System.getenv("UserProfile")) / ".mill/cache/jvm"
+      else {
+        val cacheBase = sys.env.get("XDG_CACHE_HOME").map(os.Path(_)).getOrElse(os.home / ".cache")
```

Comment:
Two things.

1. When `ctx` is defined, `ctx.env` is probably the better source for environment variables.
2. We should either use `System.getenv` or `sys.env.get` but not both.

Reply:
```scala
val env = ctx.map(_.env).getOrElse(sys.env)
```

Reply:
We can check `ctx.env` for `XDG_CACHE_HOME`, but I'm more wary of it for `UserProfile`. On Windows, env var names are case-insensitive, so that the path can be set as `Path` for example. In that case, `System.getenv("PATH")` respects the case-insensitiveness on Windows, and finds the variable. `sys.env.get("PATH")` doesn't. And IIUC, `ctx.env.get("PATH")` wouldn't either.

So I'd stick to `System.getenv` for `UserProfile`, as this is used on Windows only. But `ctx.env` and both `System.getenv` and `sys.env` are fine outside of Windows, for `XDG_CACHE_HOME`.

Reply:
Can you please add a comment, since this isn't obvious. 

Also, I'd assume Mill already suffers form this nifty details. Ideally, we have a map with case-insensitive keys semantics in `TaskCtx.env`. Also, `sys.env` seems to be a bad abstraction when it doesn't work correctly under Windows, so we should always use `System.getenv`. Nothing we can solve here, but I think we should review Mill regarding this issue.

Reply:
Looks like the case-insensitive lookup is already implemented in `sys.env`.

https://www.scala-lang.org/api/current/scala/sys.html#env-0

> An immutable Map representing the current system environment.
>
> If lookup fails, use System.getenv(_) for case-insensitive lookup on a certain platform. If that also fails, throw NoSuchElementException.

Reply:
Just checked that again in a VM to be 100% sure

<img width="1113" alt="Capture d’écran 2025-04-25 à 14 53 22" src="https://github.com/user-attachments/assets/8d2ba9c1-a355-4654-b793-e884c51ce9e7" />

The JDK sources [confirm that too](https://github.com/openjdk/jdk11u/blob/856bab316a63ce3acddd08d3fbe76b9c3223cdb6/src/java.base/windows/classes/java/lang/ProcessEnvironment.java#L209-L210).